src/containers/ArticlePage/TopicArticlePage/components/TopicArticleTaxonomy.tsx

```diff
@@ -71,93 +76,73 @@ export interface StagedTopic extends TaxonomyElement {
   metadata: TaxonomyMetadata;
 }
 
-interface State {
-  structure: StructureSubject[];
-  status: string;
-  isDirty: boolean;
-  stagedTopicChanges: StagedTopic[];
-  showWarning: boolean;
-}
-
-class TopicArticleTaxonomy extends Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      structure: [],
-      status: 'loading',
-      isDirty: false,
-      stagedTopicChanges: [],
-      showWarning: false,
-    };
-  }
-
-  componentDidMount() {
-    this.fetchTaxonomy();
-  }
-
-  componentDidUpdate({ article: { id: prevId } }: Props, prevState: State) {
-    // We need to refresh taxonomy for when an article URL has been pasted and a new article is showing
-    if (prevId !== this.props.article.id) {
-      this.fetchTaxonomy();
-    }
-  }
+const TopicArticleTaxonomy = ({ article, setIsOpen, updateNotes }: Props) => {
+  const [structure, setStructure] = useState<StructureSubject[]>([]);
+  const [status, setStatus] = useState('loading');
+  const [isDirty, setIsDirty] = useState(false);
+  const [stagedTopicChanges, setStagedTopicChanges] = useState<StagedTopic[]>([]);
+  const [showWarning, setShowWarning] = useState(false);
+  const [taxonomy, setTaxonomy] = useState<Taxonomy | undefined>(undefined);
+  const { t, i18n } = useTranslation();
+  const { userAccess } = useSession();
+
+  useEffect(() => {
+    (async () => {
+      const resources = await queryResources(article.id, article.title!.language, 'article');
+      const topics = await queryTopics(article.id, article.title!.language, 'article');
+      const localTaxonomy = { resources, topics };
+      setTaxonomy(localTaxonomy);
+      try {
+        const subjects = await fetchSubjects(i18n.language);
+
+        const sortedSubjects = subjects.filter(subject => subject.name).sort(sortByName);
+        const activeTopics = localTaxonomy.topics.filter(t => t.path) ?? [];
+        const sortedTopics = activeTopics.sort((a, b) => (a.id < b.id ? -1 : 1));
+
+        const topicConnections = await Promise.all(
+          sortedTopics.map(topic => fetchTopicConnections(topic.id)),
+        );
+
+        const topicsWithConnections = sortedTopics.map(async (topic, index) => {
+          const breadcrumb = await getBreadcrumbFromPath(topic.path, i18n.language);
+          return {
+            ...topic,
+            topicConnections: topicConnections[index],
+            breadcrumb,
+          };
+        });
+        const stagedTopicChanges = await Promise.all(topicsWithConnections);
+
+        setStatus('initial');
+        setStagedTopicChanges(stagedTopicChanges);
+        setStructure(sortedSubjects);
+      } catch (e) {
+        handleError(e);
+        setStatus('error');
+      }
+    })();
+  }, [article.id, article.title, article.title?.language, i18n.language]);
```

Comment:
Treng vi hente taksonomi ved endring av tittel?

Reply:
Nei, ikke nødvendigvis. La det til for å slippe ignore-exhaustive-deps. Det er bare side-effekten av at vi ikke har språk tilgjengelig på en enklere måte.